src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/SessionSummaryService.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service
+
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleStatus
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service.InductionScheduleService
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleStatus
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service.ReviewScheduleService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.Prisoner
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.SessionSummaryResponse
+import java.time.LocalDate
+
+private val log = KotlinLogging.logger {}
+
+@Service
+class SessionSummaryService(
+  private val prisonerSearchApiService: PrisonerSearchApiService,
+  private val reviewScheduleService: ReviewScheduleService,
+  private val inductionScheduleService: InductionScheduleService,
+) {
+
+  fun getSessionSummaries(prisonId: String): SessionSummaryResponse {
+    val today = LocalDate.now()
+    val prisoners = prisonerSearchApiService.getAllPrisonersInPrison(prisonId)
+
+    val dueReviews = mutableListOf<Prisoner>()
+    val overdueReviews = mutableListOf<Prisoner>()
+    val exemptReviews = mutableListOf<Prisoner>()
+    val dueInductions = mutableListOf<Prisoner>()
+    val overdueInductions = mutableListOf<Prisoner>()
+    val exemptInductions = mutableListOf<Prisoner>()
+
+    prisoners.forEach { prisoner ->
+      val inductionSchedule =
+        safelyGet { inductionScheduleService.getInductionScheduleForPrisoner(prisoner.prisonerNumber) }
+      val reviewSchedule =
+        safelyGet { reviewScheduleService.getLatestReviewScheduleForPrisoner(prisoner.prisonerNumber) }
+
+      if (inductionSchedule != null && (inductionSchedule.scheduleStatus != InductionScheduleStatus.COMPLETED) &&
+        reviewSchedule != null && (reviewSchedule.scheduleStatus != ReviewScheduleStatus.COMPLETED)
+      ) {
+        log.warn("Prisoner ${prisoner.prisonerNumber} has both an incomplete induction schedule and review schedule.")
+      }
+
+      inductionSchedule?.let {
+        if (inductionSchedule.scheduleStatus != InductionScheduleStatus.COMPLETED) {
+          when {
+            inductionSchedule.scheduleStatus.isExemptionOrExclusion() -> exemptInductions.add(prisoner)
```

Comment:
I can't remember how `isExemptionOrExclusion()` works (for both induction or review), but we'd not want to include exemptions through transfer/death/release/merge etc; but we would for all the "user selectable" exemptions including system outage